app/models/upload.rb

```diff
@@ -269,6 +269,14 @@ def thumbnail_height
     get_dimension(:thumbnail_height)
   end
 
+  def target_image_quality(local_path, test_quality)
+    @file_quality ||= Discourse::Utils.execute_command("identify", "-format", "%Q", local_path).to_i rescue nil
```

Comment:
If you `rescue 0` here you don't have to check `nil` below.

Reply:
👍